### ISSUE TITLE
sadf 11.2.0 produces invalid JSON output: "network" misses closing brace

### ISSUE THREAD

**Issue by andy-maier** [sysstat/sysstat#130]
Hi,
I found this other issue in sadf 11.2.0.

The "network" section misses a closing brace.

Not sure this has to do with the issue of the missing comma before the "fchosts" section, but using options that put other sections before "fchosts" do not have that issue, so I think it is actually related to "network".

```
$ sadf -j -- -n ALL 60 1
{"sysstat": {
    "sysdata-version": 3.1,
    "hosts": [
        {
            "nodename": "maiera-ub16",
            "sysname": "Linux",
            "release": "4.4.0-38-generic",
            "machine": "x86_64",
            "number-of-cpus": 4,
            "file-date": "2016-10-19",
            "file-utc-time": "08:05:01",
            "statistics": [
                {
                    "timestamp": {"date": "2016-10-19", "time": "08:15:01", "utc": 1, "interval": 600},
                    "network": {
                        "net-dev": [
                            {"iface": "lo", "rxpck": 0,00, "txpck": 0,00, "rxkB": 0,00, "txkB": 0,00, "rxcmp": 0,00, "txcmp": 0,00, "rxmcst": 0,00, "ifutil-percent": 0,00},
                            {"iface": "ens3", "rxpck": 0,57, "txpck": 0,06, "rxkB": 0,04, "txkB": 0,01, "rxcmp": 0,00, "txcmp": 0,00, "rxmcst": 0,00, "ifutil-percent": 0,00}
                        ],
                        "net-edev": [
                            {"iface": "lo", "rxerr": 0,00, "txerr": 0,00, "coll": 0,00, "rxdrop": 0,00, "txdrop": 0,00, "txcarr": 0,00, "rxfram": 0,00, "rxfifo": 0,00, "txfifo": 0,00},
                            {"iface": "ens3", "rxerr": 0,00, "txerr": 0,00, "coll": 0,00, "rxdrop": 0,00, "txdrop": 0,00, "txcarr": 0,00, "rxfram": 0,00, "rxfifo": 0,00, "txfifo": 0,00}
                        ],
                        "net-nfs": {"call": 0,00, "retrans": 0,00, "read": 0,00, "write": 0,00, "access": 0,00, "getatt": 0,00},
                        "net-nfsd": {"scall": 0,00, "badcall": 0,00, "packet": 0,00, "udp": 0,00, "tcp": 0,00, "hit": 0,00, "miss": 0,00, "sread": 0,00, "swrite": 0,00, "saccess": 0,00, "sgetatt": 0,00},
                        "net-sock": {"totsck": 580, "tcpsck": 7, "udpsck": 5, "rawsck": 0, "ip-frag": 0, "tcp-tw": 0},
                        "net-ip": {"irec": 0,06, "fwddgm": 0,00, "idel": 0,03, "orq": 0,05, "asmrq": 0,00, "asmok": 0,00, "fragok": 0,00, "fragcrt": 0,00},
                        "net-eip": {"ihdrerr": 0,00, "iadrerr": 0,00, "iukwnpr": 0,00, "idisc": 0,00, "odisc": 0,00, "onort": 0,00, "asmf": 0,00, "fragf": 0,00},
                        "net-icmp": {"imsg": 0,00, "omsg": 0,00, "iech": 0,00, "iechr": 0,00, "oech": 0,00, "oechr": 0,00, "itm": 0,00, "itmr": 0,00, "otm": 0,00, "otmr": 0,00, "iadrmk": 0,00, "iadrmkr": 0,00, "oadrmk": 0,00, "oadrmkr": 0,00},
                        "net-eicmp": {"ierr": 0,00, "oerr": 0,00, "idstunr": 0,00, "odstunr": 0,00, "itmex": 0,00, "otmex": 0,00, "iparmpb": 0,00, "oparmpb": 0,00, "isrcq": 0,00, "osrcq": 0,00, "iredir": 0,00, "oredir": 0,00},
                        "net-tcp": {"active": 0,00, "passive": 0,00, "iseg": 0,03, "oseg": 0,05},
                        "net-etcp": {"atmptf": 0,00, "estres": 0,00, "retrans": 0,00, "isegerr": 0,00, "orsts": 0,00},
                        "net-udp": {"idgm": 0,00, "odgm": 0,00, "noport": 0,00, "idgmerr": 0,00},
                        "net-sock6": {"tcp6sck": 1, "udp6sck": 3, "raw6sck": 1, "ip6-frag": 0},
                        "net-ip6": {"irec6": 0,00, "fwddgm6": 0,00, "idel6": 0,00, "orq6": 0,00, "asmrq6": 0,00, "asmok6": 0,00, "imcpck6": 0,00, "omcpck6": 0,00, "fragok6": 0,00, "fragcr6": 0,00},
                        "net-eip6": {"ihdrer6": 0,00, "iadrer6": 0,00, "iukwnp6": 0,00, "i2big6": 0,00, "idisc6": 0,00, "odisc6": 0,00, "inort6": 0,00, "onort6": 0,00, "asmf6": 0,00, "fragf6": 0,00, "itrpck6": 0,00},
                        "net-icmp6": {"imsg6": 0,00, "omsg6": 0,00, "iech6": 0,00, "iechr6": 0,00, "oechr6": 0,00, "igmbq6": 0,00, "igmbr6": 0,00, "ogmbr6": 0,00, "igmbrd6": 0,00, "ogmbrd6": 0,00, "irtsol6": 0,00, "ortsol6": 0,00, "irtad6": 0,00, "inbsol6": 0,00, "onbsol6": 0,00, "inbad6": 0,00, "onbad6": 0,00},
                        "net-eicmp6": {"ierr6": 0,00, "idtunr6": 0,00, "odtunr6": 0,00, "itmex6": 0,00, "otmex6": 0,00, "iprmpb6": 0,00, "oprmpb6": 0,00, "iredir6": 0,00, "oredir6": 0,00, "ipck2b6": 0,00, "opck2b6": 0,00},
                        "net-udp6": {"idgm6": 0,00, "odgm6": 0,00, "noport6": 0,00, "idgmer6": 0,00}                    "fchosts": [

                    ]
                }
            ],
            "restarts": [
            ]
        }
    ]
}}
```

**Comment by andy-maier** [comment]
I verified that 10.1.5 does not have this issue.

**Comment by andy-maier** [comment]
The closing brace is also missing in the example shown in issue #90, but I guess it went by undetected because the focus was on the missing comma before "fchosts".

**Comment by sysstat** [comment]
Hi,
Can you check whether the bug still exists with the latest 11.2.x stable version (11.2.7 as of this writing)?
I think this problem was fixed in version 11.2.1 with commit 54b03d1.
Thanks.

**Comment by andy-maier** [comment]
I will. give me some time (couple days).
